ecosystem/sep-0039.md

```diff
@@ -0,0 +1,193 @@
+## Preamble
+
+```
+SEP: 0039
+Title: On-Chain Storage of Non-Fungible Assets
+Author: George Kudrayvtsev (@Shaptic)
+Track: Standard
+Status: Draft
+Created: 2021-11-22
+Discussion: TODO
+```
+
+
+## Simple Summary
+This SEP describes a specific way to store arbitrary data on-chain with the goal of standardizing the way NFTs are represented in the Stellar ledger.
+
+
+## Dependencies
+While not an explicit dependency, there may be a future CAP that modifies the protocol to allow a 128-byte row of ManageData to be arbitrarily divided into the string "key" and binary "data" portions, as opposed to the current model of having strict 64-byte halves.
+
+With that CAP, the data model introduced [below](#specification) could/should change to be friendlier and more optimized.
```

Comment:
I think the value that this delivers would greatly simplify this SEP, and the fact the CAP would be very small both in specification and implementation makes it attractive as a quick win that'd also reduce the space NFTs consume in stellar-core instances. It'd remove the need for base91 encoding from this SEP, which while neat and the best thing we can do under the current circumstances, is also unfortunate. @Shaptic Are you planning to write a CAP for this idea?